x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryTranslator.java

```diff
@@ -275,8 +276,11 @@ else if (exp instanceof GroupingFunction) {
                             Expression field = h.field();
 
                             // date histogram
-                            if (h.dataType() == DataType.DATETIME) {
+                            if (h.dataType().isDateBased()) {
                                 long intervalAsMillis = Intervals.inMillis(h.interval());
+                                if (h.dataType() == DATE) {
+                                    intervalAsMillis = DateUtils.minDayInterval(intervalAsMillis);
```

Comment:
I'm assuming these two lines here cover the scenario described in the documentation here https://github.com/elastic/elasticsearch/pull/37693/files#diff-fd0095853b8c10e4fd39bd7ae4958aabR80. Can you, please, add a small comment in code for this?